server/resource_server/controller/controller_test.go

```diff
@@ -331,49 +344,128 @@ func TestGetClientData(t *testing.T) {
 	assert.Equal(t, "https://gcitizen.com/callback", response.RedirectURI)
 }
 
-func TestVerifyEmailHandler(t *testing.T) {
-	// Generate a Mock JWT token
-	tokenString, err := utils.GenerateToken(models.User{
-		ID:       1,
-		Username: "test_user",
-	})
+func TestVerifyEmailHandler_FailedToVerifyEmail(t *testing.T) {
+	req, err := http.NewRequest("GET", "/api/v1/verify-email", nil)
 	if err != nil {
 		t.Fatal(err)
 	}
+	req.Header.Set("Authorization", "Bearer "+authenticationToken)
 
-	// Create a mock request
+	mockService := &MockService{
+		verifyEmail: func(userID uint) error {
+			return fmt.Errorf("failed to verify email for user %d", userID)
+		},
+	}
+	req = req.WithContext(context.WithValue(req.Context(), "service", mockService))
+
+	rr := httptest.NewRecorder()
+
+	Ctl.VerifyEmailHandler(rr, req)
+
+	assert.Equal(t, http.StatusBadRequest, rr.Code)
+
+	var response utils.Response
+	if err := json.NewDecoder(rr.Body).Decode(&response); err != nil {
+		t.Fatal(err)
+	}
+
+	assert.False(t, response.Status)
+	assert.Equal(t, "Failed to verify email", response.Message)
+	assert.NotNil(t, response.Error)
+}
+
+func TestVerifyEmailHandler_Success(t *testing.T) {
 	req, err := http.NewRequest("GET", "/api/v1/verify-email", nil)
 	if err != nil {
 		t.Fatal(err)
 	}
+	req.Header.Set("Authorization", "Bearer "+authenticationToken)
 
-	// Set the Authorization header
-	req.Header.Set("Authorization", "Bearer "+tokenString)
-
-	// Create a mock service and set it in the request context
-	mockService := &MockService{}
+	mockService := &MockService{
+		verifyEmail: func(userID uint) error {
+			return nil
+		},
+	}
 	req = req.WithContext(context.WithValue(req.Context(), "service", mockService))
 
-	// Create a ResponseRecorder to record the response
 	rr := httptest.NewRecorder()
 
-	// Call the handler function
 	Ctl.VerifyEmailHandler(rr, req)
 
-	// Check the status code
 	assert.Equal(t, http.StatusOK, rr.Code)
 
-	// Decode the response body
 	var response utils.Response
 	if err := json.NewDecoder(rr.Body).Decode(&response); err != nil {
 		t.Fatal(err)
 	}
 
-	// Now assert the fields directly
 	assert.True(t, response.Status)
 	assert.Equal(t, "OK!", response.Message)
+	assert.Equal(t, "Verification email sent", response.Data)
+	assert.Nil(t, response.Error)
+}
+
+func TestCheckEmailVerificationCode_FailedToVerifyCode(t *testing.T) {
+	requestBody := []byte(`{"code": "123467"}`)
```

Comment:
Looking at the function CheckEmailVerificationCode(), I see a few more opportunities for testing:
1) Token fails validation (tests error handling in Line 257)
2) A malformed request body (tests error handling on Line 255)